### DISCUSSION TITLE
Make DatasyncClientData properties virtual

### DISCUSSION THREAD

**Discussion by scout208** [Azure/azure-mobile-apps#720] [Ideas]
The `EntityTableData` class in `Microsoft.AspNetCore.Datasync.EFCore` has all its properties marked as `virtual`. This is nice since it give the option to override them if you want a different implementation of any of them than the default. However, the client class (`Microsoft.Datasync.Client.DataSyncClient`) has non-virtual properties so there is no option to override these. Was there a particular reason for this?

I want to override the `UpdatedAt` property on the client side so that it is set to the current time instead of `DateTimeOffset.MinValue` which it is by default.

**Comment by adrianhall** [comment]
On the client side, you don't want to set UpdatedAt at all - it's explicitly set by the server side.  This is why the values in DataSyncTableData are not virtual.

**Comment by scout208** [comment]
Okay, that makes sense. In my scenario I'm wanting to maintain when the item was updated on the client, which for an offline client could be different than the time the item reached the server. I'll just add another property to keep track of that. Closing this discussion.
